generators/python/core_utilities/shared/unchecked_base_model.py

```diff
@@ -128,6 +128,15 @@ def _convert_undiscriminated_union_type(union_type: typing.Type[typing.Any], obj
         return object_
 
     for inner_type in inner_types:
+        # Handle lists of objects that need parsing
+        if get_origin(inner_type) == list and isinstance(object_, list):
```

Comment:
```suggestion
        if get_origin(inner_type) is list and isinstance(object_, list):
```

nit: type comparison by `is` (as with other uses of `get_origin()`)